quipucords/scanner/vcenter/inspect.py

```diff
@@ -133,11 +133,14 @@ def parse_cluster_props(self, props, parents_dict):
             if prop.name == 'name':
                 facts['cluster.name'] = prop.val
             elif prop.name == 'parent':
-                parent = parents_dict[str(prop.val)]
-                while parent['type'] != 'vim.Datacenter':
-                    parent = parents_dict[parent['parent']]
-                facts['cluster.datacenter'] = parent['name']
-
+                parent = parents_dict.get(str(prop.val))
+                while parent.get('type') != 'vim.Datacenter':
+                    if parent is not None:
```

Comment:
Parent here will always be not None because the while condition as it is will raise an exception if parent is None, so it won't get here if parent is None.